### ISSUE TITLE
Missing SSL Certificate required by Google Auth after Authentication

### ISSUE THREAD

**Issue by carlosparadis** [erdl/survey_admin#51]
This issue is consequence of #41 

# Example Scenario

If someone attempts to login to survey_admin via Apache HTTP, Google will authenticate, and then it will attempt to redirect the user to whatever place you (Jason) chose on Google's control panel, where the API Key was generated for Google oAuth. However, the chosen URL at the control panel that the user will be redirected to will require `SSL`, therefore throwing the error. 

This can be bypass by doing:

https://github.com/erdl/survey_admin/blob/c18233a05a1038426027bb06cea0e7ae3f6e1e30/run.py#L5

However, the effects of such band-aid solution are not fully known and were not moved to production. 

# Current Solutions

 * We should be able to request an SSL Certificate for free [from ITS Server](https://www.hawaii.edu/sitelic/incommon/vm_ssl.html). However, some of the required information is unclear, and would likely require @eileenpeppard to be involved to make a request. Need to look further into this. 
  * SOEST may also offer us some service, as we own a DNS from them which is used by survey_admin, namely csbcd.soest.hawaii.edu. Information related to SOEST IP can be found an unrelated matter e-mail that Carlos and Eileen both have titled `Re: Getting started with MATLAB license`. [SOEST IT page](https://www.soest.hawaii.edu/rcf/faq.htm), unfortunately, does not offer much information. See [this page](https://its.unl.edu/services/ssl-certificates) for the meaning about types of certificates. 

# Reference Material

Two simple videos in layman terms:

 * (2 mins) https://www.youtube.com/watch?v=SJJmoDZ3il8
 * (8 mins) https://www.youtube.com/watch?v=earzZpX-PiY&feature=youtu.be&t=7m8s 

 * http://oauthlib.readthedocs.io/en/latest/oauth2/security.html
 * (bottom of page 21) https://media.readthedocs.org/pdf/requests-oauthlib/latest/requests-oauthlib.pdf

**Comment by carlosparadis** [comment]
@ryantanaka did you hear anything from Justin on this front? It's been over a week now.

**Comment by ryantanaka** [comment]
@carlosparadis  nope, just asked eileen and she hasn't heard back as well, but  i  we will see what he says in response to your email

**Comment by carlosparadis** [comment]
@ryantanaka sounds good.

**Comment by carlosparadis** [comment]
@ryantanaka This is taking too long. If Justin doesn't follow-up my e-mail, could you take the lead on this and contact the department he said he would?

**Comment by ryantanaka** [comment]
@ryantanaka

```
# ONLY FOR TESTING
#import os
#os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
```

**Comment by carlosparadis** [comment]
@ryantanaka 2 days for 2 months on this 🍶

**Comment by ryantanaka** [comment]
Current Status:

Our SSL certificate was issued through the [InCommon 1 Year SSL Certificate Program](https://www.hawaii.edu/sitelic/incommon/vm_ssl.html) on May 4, 2018 and will require renewal the following year. 

Justin Delp requested this certificate through ITS and is the technical contact listed on the certificate. He did some preliminary setup on the server, and I dropped in the necessary certificate files in the appropriate places when they were received. 

The server now uses https for both `survey_display` and `survey_admin`. All features appear to be working. Today I will be going through both apps to do some testing to make sure that all of their features are still working properly. 

The following blurb of code is no longer used in `/var/www/survey_admin/survey_admin/__init__.py` since HTTPS has been set up.
```
# ONLY FOR TESTING
#import os
#os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
```

The appropriate changes were then made to `/var/www/survey_admin/config.json` to reflect the changes made in the apache web server. Appropriate changes were also made in the OAuth dashboard, which is managed through Google Cloud Platform. 

~~Justin Delp is looking into requiring https when going to our web apps. This way, when someone forgets to use `https` in any of the urls, they get redirected to the  same url but with `https`.~~ This has been completed. Apache will redirect to the the secure site.
